### ISSUE TITLE
[JAXRS-CXF] Enhancement: generate builder-style methods in models

### ISSUE THREAD

**Issue by MatanRubin** [swagger-api/swagger-codegen#4386]
Currently, JAXRS-CXF codegen generates getters and setters for model members.
The Java code generator has the nice feature of generating builder-style methods as well.
The suggestion is to generate builder style methods in JAXRS-CXF as well.

Making this change would allow us to get rid of cummbersome code like this:

```
MyResponse response = new MyResponse();
reponse.setName("Mickey");
response.setLastName("Mouse");
response.setAge(89);
return response;
```

For this:

```
return new MyResponse().name("Mickey").lastName("Mouse").age(89);
```

Swagger-codegen version: 2.2.1

I can contribute this fix myself, but wanted to make sure first that it's something we want.
@wing328, let me know if you want me to work on this.

**Comment by wing328** [comment]
@MatanRubin yes, I think that will be a nice enhancement. Let us know if you need any help with the PR.

**Comment by MatanRubin** [comment]
Thanks for the quick reply @wing328.
Come to think about it, is there a reason why JAXRS (and all other JAVA related generators) couldn't share models with the JAVA generator?

**Comment by wing328** [comment]
@MatanRubin yes, ideally that's what we want - reuse models and other share classes by all Java-related generators but that would require quite an amount of work to do the refactoring.

**Comment by jfiala** [comment]
@wing328 I just came across this discussion, I suggested using centralized templates for beanvalidation initially but as I remember @fehguy preferred to keep the shared classes/templates separately. I think it would definitely make sense to keep centralized templates and use lambdas to keeep everything consistent. Pls let me know if I should make the beanvalidation templates shared?

**Comment by jfiala** [comment]
@wing328 For CXF this already seems to be implemented, the next step is to port this to all Java languages!? Should we open a separate issue or add the language checklist here?

**Comment by wing328** [comment]
>  lambdas to keep everything consistent.

@fehguy had bad experience with lambdas before. If you think `lambdas` will be beneficial to the project, then we can restart the conversation again.

> Should we open a separate issue or add the language checklist here?

I prefer a new issue with checklist instead of reusing this one.

**Comment by jfiala** [comment]
@wing328  @MatanRubin I think we can close this now as it has been implemented for cxf, further language porting see #4938, lambdas see #4937.
